### PR TITLE
handling archive suborders from table

### PR DESCRIPTION
**Pull request by mazenmmbadr**

### CODE REVIEW

File: ui_testing/pages/orders_page.py [hunk 1]
```diff
@@ -80,11 +80,6 @@ def filter_by_order_no(self, filter_text):
         self.filter_by(filter_element='orders:filter_order_no', filter_text=filter_text, field_type='text')
         self.filter_apply()
 
-    def open_filter_menu(self):
-        filter = self.base_selenium.find_element_in_element(source_element='general:menu_filter_view',
```
[the remark this answers is not in the document]
Comment (dianaeldemerdash) [reply ui_testing/pages/orders_page.py]: done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 2]
```diff
@@ -172,28 +174,30 @@ def test004_archive_order(self):
...
-        for selected_order_data in selected_orders_data:
-            self.base_selenium.LOGGER.info(' + Order with analysis number =  {} restored successfully?'.format(
-                selected_order_data['Analysis No.']))
-            self.assertTrue(self.order_page.is_order_exist(
-                value=selected_order_data['Analysis No.']))
+        all_orders = self.orders_api.get_archived_orders_json().json()['orders']
+        record_id = randint(0, len(all_orders) - 2)
+        order_data = all_orders[record_id]
+
+        order_no = order_data['orderNo']
```
[the remark this answers is not in the document]
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 3]
```diff
@@ -172,28 +174,30 @@ def test004_archive_order(self):
...
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no, field_type='text')
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        self.order_page.restore_table_suborder(index=0)
```
[the remark this answers is not in the document]
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: @ghada-37 this test case should test when I restore the suborder so I updated the test case it self on Jira to do that

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 4]
```diff
@@ -1196,126 +1113,126 @@ def test027_update_material_type(self):
     # will continue with us & then put the test case number for it 
     def test026_update_suborder_article(self):
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:1114]: why did you commented this test case?

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 5]
```diff
@@ -1593,37 +1510,37 @@ def test031_user_can_add_suborder_with_test_units(self):
         self.analyses_page.get_analyses_page()
         self.base_selenium.LOGGER.info('Filter by order no to make sure that the analysis was not deleted')
 
-        analysis_records = self.analyses_page.search(value=order_no)
-        analysis_count = len(analysis_records) - 1
+        #analysis_records = self.analyses_page.search(value=order_no)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:1513]: why did u commented it

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 6]
```diff
@@ -189,28 +188,28 @@ def test004_archive_order(self):
...
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        self.order_page.restore_table_suborder(index=0)
+        self.base_selenium.LOGGER.info('make sure that suborder is restored successfully')
+        if len(suborders_data) > 1:
+            suborder_data_after_restore = self.order_page.get_table_data()
+            self.assertNotEqual(suborder_data_after_restore[0], suborders_data[0])
+        else:
+            table_records_count = len(self.order_page.result_table()) -1
+            self.assertEqual(table_records_count, 0)
+
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:212]: I think you should assert that suborder restored by checking that it is in active table

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 7]
```diff
@@ -970,109 +969,28 @@ def test023_create_existing_order_with_test_units_and_change_article(self):
...
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders()
+        order_data = random.choice(orders['orders'])
+
+        order_no = order_data['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no, field_type='text')
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        self.order_page.archive_table_suborder(index=0)
+        self.base_selenium.LOGGER.info('make sure that suborder is archived successfully')
+        if len(suborders_data) > 1:
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:987]: to make sure that suborder is archived successfully, you should assert it's in archived table or to search for it in analysis active table

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:204]: ok

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 6 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:217]: if order has another suborder, this assertion will be true even if suborder not restored
so, you need to make assertion by analysis no not order no

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 8]
```diff
@@ -970,109 +974,32 @@ def test023_create_existing_order_with_test_units_and_change_article(self):
...
-        self.orders_page.filter_by_analysis_number(filter_text=analysis_no)
-
-        rows_count = self.order_page.result_table()
-        self.base_selenium.LOGGER.info(
-            '+ Assert count archived orders with a specific analysis number: {}'.format(len(rows_count)))
-
-        self.assertEqual(len(rows_count) - 1, 1)
+        self.order_page.search(order_no)
+        results = self.order_page.result_table()[0].text
+        self.assertIn(order_no.replace("'", ""), results.replace("'", ""))
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:1002]: the same here if order has multiple suborders and one of them was archived before this assertion will raise true even if suborder not archived
you need to search by analysis no

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 9]
```diff
@@ -1406,7 +1333,7 @@ def test026_update_suborder_article(self):
     ### SYNTAX ERROR ###
 
     # will continue with us apply it from the second suborder & need test case number for it to apply from the second suborder
-    @parameterized.expand(['save_btn', 'cancel'])
+   # @parameterized.expand(['save_btn', 'cancel'])
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test06_orders.py:1336]: why did u commented this?

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 8 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py:1002]: @ghada-37 I will assert with the analysis number in the two test case and abut the previous comment ( if order has another suborder, this assertion will be true even if suborder not restored
so, you need to make assertion by analysis no not order no ) it the same comment you mean right

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:1002]: it's the same comment yeah., assertion or search whatever the method you select it should depend on anaysis no

**Comment by 0xIslamTaha** [comment]
@dianaeldemerdash there is code which is not matching with the latest update in master and there is a lot of comments which isn't done too so please double-check it again

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 4 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: api_testing/apis/orders_api.py [hunk 10]
```diff
@@ -211,5 +211,11 @@ def _format_payload(payload):
         elif payload['yearOption'] == 2:
             payload['orderNoWithYear'] = payload['year'] + '-' + payload['orderNo']
 
-        payload['materialTypeId'] = payload['materialType']['id']
+        request_body['materialTypeId'] = kwargs['materialType']['id']
```
[the remark this answers is not in the document]
Comment (dianaeldemerdash) [reply api_testing/apis/orders_api.py]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 5 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 6 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py:212]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 7 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 8 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 9 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

**Comment by dianaeldemerdash** [comment]
@0xIslamTaha I wrote that all the comments are done

**Comment by 0xIslamTaha** [comment]
> @0xIslamTaha I wrote that all the comments are done

@dianaeldemerdash PLEASE CHECK MY COMMENTS BEFORE YOU REPLAY.  

here is an example u just wrote "done" but the code still wrong
image

File: .travis.yml [hunk 11]
```diff
@@ -27,5 +27,21 @@ script:
...
+  #- xvfb-run -a nosetests-3.4 -vs --logging-level=WARNING --tc-file=config.ini ui_testing/testcases/basic_tests/test02_articles.py:ArticlesTestCases.test024_archive_optional_config_fields_effect__1_create
+  #- xvfb-run -a nosetests-3.4 -vs --logging-level=WARNING --tc-file=config.ini ui_testing/testcases/basic_tests/test02_articles.py:ArticlesTestCases.test025_restore_optional_config_fields_effect__0_edit
+  #- xvfb-run -a nosetests-3.4 -vs --logging-level=WARNING --tc-file=config.ini ui_testing/testcases/basic_tests/test02_articles.py:ArticlesTestCases.test025_restore_optional_config_fields_effect__1_create
+  #- xvfb-run -a nosetests-3.4 -vs --logging-level=WARNING --tc-file=config.ini ui_testing/testcases/basic_tests/test02_articles.py:ArticlesTestCases.test026_archive_optional_config_fields_does_not_effect_table
+ # - xvfb-run -a nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test034_Duplicate_sub_order_and_cahange_materiel_type
```
Comment (0xIslamTaha) [inline .travis.yml:46]: Remove

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 6 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test06_orders.py:212]: @deenaahmed `self.assertTrue(ANY_VALID_VALUE)` is true so u need to check that `result` data is matching with the suborder data

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 12]
```diff
@@ -972,109 +970,27 @@ def test023_create_existing_order_with_test_units_and_change_article(self):
...
-
-        rows_count = self.order_page.result_table()
-        self.base_selenium.LOGGER.info(
-            '+ Assert count archived orders with a specific analysis number: {}'.format(len(rows_count)))
-
-        self.assertEqual(len(rows_count) - 1, 1)
+        row = self.order_page.search(suborders_data['Analysis No.'])[0]
+        result= self.base_selenium.get_row_cells_dict_related_to_header(
+            row=row, table_element='general:table_child')
+        self.assertTrue(row, result)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:993]: same comment

Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test06_orders.py]: Done

File: .travis.yml (hunk 11 above)
Comment (dianaeldemerdash) [reply .travis.yml]: Done

**Pull request merged**